source/lambda/apiprocessor/redaction.py

```diff
@@ -271,7 +265,12 @@ def saveDocumentRedaction(documentId, bucket, table, body):
         
         if statusCode != 200:
             return (statusCode, result)
-    
+
+    # validate there aren't any other keys that the requires ones
+    if len(redaction.keys()) != REDACTION_NUM_OF_KEYS:
```

Comment:
Since we know the exact keys that can be passed, why dont we check if redacts.keys() has the exact values we need.
```
 if not (all(x in ["uuid","headers","footers","redactedItems"] for x in redaction.keys() )):
   return "Please only provide uuid, headers, footers and redactedItems")
```
and 
```
if len(redaction.keys())!=len(set(redaction.keys)):
   return "No duplicate keys allowed"

```

Reply:
the first snippet didn't work as expected, so replaced it.   Corrected the second snippet as it was missing ()  on redaction.keys